api_tests/metrics/test_raw_metrics.py

```diff
@@ -0,0 +1,139 @@
+
+    def test_put(self, app, user, base_url):
+        put_return = {
+            '_index': 'customer',
+            '_type': '_doc',
+            '_id': '1',
+            '_version': 1,
+            'result': 'created',
+            '_shards': {
+                'total': 2,
+                'successful': 1,
+                'failed': 0
+            },
+            '_seq_no': 0,
+            '_primary_term': 1
+        }
+
+        put_url = '{}customer/_doc/1'.format(base_url)
+        put_data = {
+            'name': 'John Doe'
+        }
+        res = app.put_json_api(put_url, put_data, auth=user.auth)
+        assert res.json == put_return
+
+    def test_put_no_perms(self, app, other_user, base_url):
+        put_url = '{}customer/_doc/1'.format(base_url)
+        put_data = {
+            'name': 'John Doe'
+        }
+        res = app.put_json_api(put_url, put_data, auth=other_user.auth, expect_errors=True)
+        assert res.status_code == 403
+        assert res.json['errors'][0]['detail'] == 'You do not have permission to perform this action.'
+
+    def test_post(self, app, user, base_url):
+        post_return = {
+            '_index': 'customer',
+            '_type': '_doc',
+            '_id': '1',
+            '_version': 1,
+            'result': 'created',
+            '_shards': {
+                'total': 2,
+                'successful': 1,
+                'failed': 0
+            },
+            '_seq_no': 0,
+            '_primary_term': 1
+        }
+
+        post_url = '{}customer/_doc/1'.format(base_url)
+        post_data = {
+            'name': 'Jane Doe'
+        }
+        res = app.post_json_api(post_url, post_data, auth=user.auth)
+        assert res.json == post_return
+
+    def test_post_no_perms(self, app, other_user, base_url):
+        post_url = '{}customer/_doc/1'.format(base_url)
+        post_data = {
+            'name': 'John Doe'
+        }
+        res = app.post_json_api(post_url, post_data, auth=other_user.auth, expect_errors=True)
+        assert res.status_code == 403
+        assert res.json['errors'][0]['detail'] == 'You do not have permission to perform this action.'
+
+    def test_post_and_get(self, app, user, base_url):
+        post_return = {
+            '_index': 'customer',
+            '_type': '_doc',
+            '_id': '1',
+            '_version': 1,
+            'result': 'created',
+            '_shards': {
+                'total': 2,
+                'successful': 1,
+                'failed': 0
+            },
+            '_seq_no': 0,
+            '_primary_term': 1
+        }
+
+        post_url = '{}customer/_doc/1'.format(base_url)
+        post_data = {
+            'name': 'Beyonce'
+        }
+        res = app.post_json_api(post_url, post_data, auth=user.auth)
+        assert res.json == post_return
+
+        time.sleep(3)
+
+        get_url = '{}_search?q=*'.format(base_url)
```

Comment:
The only downside of this test is that it would have passed before when we weren't using the query parameters, yes? I forget, but does elastic return the query information in the response so you can make sure it went through?

Reply:
I had to alter this test to include that search parameter. `_search?q=*` does the same thing as `connection.search()`. Its just a blank query. Without `_search?q=*`, it returns stats about elastic search.

Reply:
I added an additional GET with assertion to make sure the query params are adhered to.